addons/product/models/product_pricelist.py

```diff
@@ -221,7 +221,8 @@ def _compute_price_rule(self, products_qty_partner, date=False, uom_id=False):
                     price = product.price_compute(rule.base)[product.id]
 
                 if price is not False:
-                    price = rule._compute_price(price, price_uom, product, quantity=qty, partner=partner)
+                    rule_with_date = rule.with_context({'date': date})
```

Comment:
```suggestion
                    # pass the date through the context for further currency conversions
                    rule_with_date_context = rule.with_context(date=date)
```
FYI, `with_context(dict)` returns a recordset with its whole context replace by the provided dict.
In most cases, you don't want to reset the whole context (which might contain interesting information you have to keep: lang, tz, companies, ...) so you should mostly use the other way and only provide the keys you want to modify in the existing context ;).